yt_dlp/extractor/youtube.py

```diff
@@ -857,6 +857,18 @@ def generate_api_headers(
         }
         return filter_dict(headers)
 
+    def _download_webpage_with_retries(self, *args, retry_fatal=False, retry_on_status=None, **kwargs):
+        for retry in self.RetryManager(fatal=retry_fatal):
+            try:
+                return self._download_webpage(*args, **kwargs)
+            except ExtractorError as e:
+                if isinstance(e.cause, network_exceptions):
+                    if not isinstance(e.cause, HTTPError) or e.cause.status not in (retry_on_status or (403, 429)):
```

Comment:
should we exclude `400` too? since it usually means broken client config

Reply:
note this is a webpage request, not innertube. technically all 4XX codes should not be retried in I think.